scripts/build

```diff
@@ -373,13 +373,14 @@ fi
 
 pkg_call_exists post_make_$TARGET && pkg_call post_make_$TARGET
 
-# make install writes directly to sysroot which then needs fixing up - this is racy
-if listcontains "configure:target cmake-make:target autotools:target make:target" "${PKG_TOOLCHAIN}:${TARGET}"; then
-  acquire_exclusive_lock "${PKG_NAME}:${TARGET}" "build"
-  PKG_NEED_UNLOCK=yes
-else
-  PKG_NEED_UNLOCK=no
-fi
+# Hack around directly writing/modifying the content of a shared sysroot
+# by temporarily installing new files to a package specific sysroot
+PKG_ORIG_SYSROOT_PREFIX="${SYSROOT_PREFIX}"
+export SYSROOT_PREFIX="${BUILD}/.sysroot/${PKG_NAME}.${TARGET}"
+rm -fr "${SYSROOT_PREFIX}"
+for d in /usr/lib /usr/include /usr/bin /usr/lib/pkgconfig; do
```

Comment:
what is the purpose of this for loop? Do some packages fail to install if the directory doesn't exist already?

Reply:
Yes, basically a number of packages currently work only because a preceding package has already created the directories in the shared sysroot, so if we now give each package an isolated (and empty) sysroot then those directories won't exist, so the packages fail.

One solution would be to modify every package so that it creates the `usr/lib`, `usr/include`,  `usr/bin` etc. directories as required (a lot of tedious work), or we just pre-create these commonly-used directories and call it a day (plus, I'm lazy). :smile:

Fixing packages so that we can remove the `mkdir` loop is probably something for another day.

Reply:
I'm fine with that but I think it would be good to know what packages do fail because that is a bug on their end not ours. Maybe add a `# todo` here so we know to check back on this later 👍

Reply:
TODO added